### DISCUSSION TITLE
How to write the associativity rule?

### DISCUSSION THREAD

**Discussion by ghost** [tree-sitter/tree-sitter#1825] [Q&A]
Given a syntax with ebnf:
```ebnf
Expr ::= SubExpr (Operator SubExpr)?
```
I wrote a rule as:
```js
Expr: ($) => seq($.SubExpr, optional(seq($.Operator, $.SubExpr))),
```
the tree-sitter generator failed with:
```sh
Unresolved conflict for symbol sequence:

  'def'  Identifier  '('  ')'  '{'  'for'  '('  Identifier  '='  SubExpr  •  '<'  …

Possible interpretations:

  1:  'def'  Identifier  '('  ')'  '{'  'for'  '('  Identifier  '='  (Expr  SubExpr  •  Operator  SubExpr)
  2:  'def'  Identifier  '('  ')'  '{'  'for'  '('  Identifier  '='  (Expr  SubExpr)  •  '<'  …

Possible resolutions:

  1:  Specify a left or right associativity in `Expr`
  2:  Add a conflict for these rules: `Expr`
```
How can I write the associativity rule for option 1 or 2? 
I just wrapped the `seq(...)` with `prec.[left or right](seq(...))`, seems not work at all.

**Comment by ahelwer** [comment]
What happens if you write it as `Expr: ($) => prec.left(seq($.SubExpr, optional(seq($.Operator, $.SubExpr)))),`? Same conflict or different?

**Reply by ghost** [reply]
No conflict anymore. I tried this kind of wrapping, but then the expr cannot be parsed correctly.

**Reply by ahelwer** [reply]
Okay, so how do you want it to be parsed then?
